src/config.rs

```diff
@@ -405,6 +406,22 @@ impl Cfg {
         Ok(())
     }
 
+    pub fn set_auto_self_update(&mut self, mode: &str) -> Result<()> {
```

Comment:
I'm not entirely sure I like this use of strings here, I think I'd prefer an enumeration for the mode, but I also won't block this PR on that if you don't want to rework our internal APIs to use an enum.

Reply:
Yes, I totally agree that using enumeration is better. I added a fixme and I will change our code to support enums in the next PR.